src/content/1.7/development/Multistore/shop-context/_index.md

```diff
@@ -0,0 +1,81 @@
+title: Shops context
+menuTitle: Shops context
+summary: "How to get the current shop context"
+weight: 20
+---
+
+# Shop context
+
+In many cases, your code will need to know if the multistore mode is enabled, if there are more than one shops activated, and if yes, what is the current shop context. This is where the shop context and multistore feature adapters will come in handy.
+
+## The multistore feature adapter
+
+**Service identifier:** `prestashop.adapter.feature.multistore` (it takes the legacy configuration adapter as a parameter) <br>
+**Class:** `PrestaShop\PrestaShop\Adapter\Feature\MultistoreFeature`
+
+
+The multistore feature adapter can be used to:
+
+- detect if multistore was enabled in the configuration,
+
+- detect if multistore is active (if it is enabled, and if there are more than one active shops),
+
+- enable/disable the multistore feature.
+
+Example:
+
+```php
+<?php
+
+// Get the multistore feature adapter from the service container
+$multistoreFeature = $this->get('prestashop.adapter.feature.multistore');
+
+// Is the multistore feature enabled?
+$isMultistoreEnabled = $multistoreFeature->isActive();
+
+// This will check that multistore is enabled AND that there are at least two active shops
+$isMultistoreUsed = $multistoreFeature->isUsed();
+
+// Disable / enable multistore feature
+$multistoreFeature->disable();
+$multistoreFeature->enable();
+
+```
+
+## The shop context adapter
+
+**Service identifier:** `prestashop.adapter.shop.context` <br>
+**Class:** `PrestaShop\PrestaShop\Adapter\Shop\Context`
+
+When the multistore feature is enabled, at the top of all the back office pages, a dropdown appears, it allows selecting the shop context we want (all shops, a group shop, a single shop...). The shop context adapter will tell you in what context your code is operating.
+
+Here are some examples of commonly used functions in the shop context adapter:
+
+```php
+<?php
+
+// Get the shop context adapter from the service container
+$shopContextAdapter = $this->get('prestashop.adapter.shop.context');
+
+// Get a list of all active shops
+$shopList = $shopContext->getShops(false, true);
+
+// Is the current context a group of shops ?
+$isShopGroupContext = $shopContext->isShopGroupContext();
+
+// Is the current context a single shop ?
+$isSingleShopContext = $shopContext->isSingleShopContext();
+
+// Is "all shops" context
+$isAllShopContext = $shopContext->isAllContext();
+
+
+// Get a list of IDs for all shops belonging to the current context (useful is the current context is a group) 
+$shopList = $shopContext->getContextListShopID();
+
+// Get the id of the shop in the current context (useful if the current context is a single shop)
+$currentIdShop = $shopContext->getContextShopID();
+```
+
+The shop context adapter class has many functions with self-explanatory naming and helpful comments, feel free to [explore it by yourself](https://github.com/PrestaShop/PrestaShop/blob/1.7.7.x/src/Adapter/Shop/Context.php).
```

Comment:
```suggestion
The shop context adapter class has many functions with self-explanatory naming and helpful comments, feel free to [explore it by yourself](https://github.com/PrestaShop/PrestaShop/blob/develop/src/Adapter/Shop/Context.php).
```

Not sure if we should link to specific version.

Reply:
Good question, @matks what do you think ? I thought since we are in 1.7.7 documentation it was better to link the 1.7.7 code, because the develop branch is ever changing.

Reply:
It's global documentation, right? And this feature you're documenting in some places will work from 1.7.8, I think `develop` is better choice, also, if we change something on develop it should be documented anyway

Reply:
true